File: app/src/main/java/com/usharik/app/CzechCase.java
package com.usharik.app;

public enum CzechCase {
    NOMINATIV(0, "Nominativ", "kdo? co?"),
    GENITIV(1, "Genitiv", "koho? čeho?"),
    DATIV(2, "Dativ", "komu? čemu?"),
    AKUZATIV(3, "Akuzativ", "koho? co?"),
    VOKATIV(4, "Vokativ", "-"),
    LOKAL(5, "Lokál", "(o) kom? (o) čem?"),
    INSTRUMENTAL(6, "Instrumentál", "kým? čím?");

    public final int index;
    public final String name;
    public final String question;

    CzechCase(int index, String name, String question) {
        this.index = index;
        this.name = name;
        this.question = question;
    }

    public static CzechCase fromIndex(int index) {
        for (CzechCase c : values()) {
            if (c.index == index) return c;
        }
        throw new IllegalArgumentException("Invalid case index: " + index + ". Valid range is 0-6.");
    }
}

File: app/src/main/kotlin/com/usharik/app/CzechCase.kt
package com.usharik.app

enum class CzechCase(
    @JvmField val index: Int,
    @JvmField val displayName: String,
    @JvmField val helperWord: String,
    @JvmField val question: String,
) {
    NOMINATIV(0, "Nominativ", "", "kdo? co?"),
    GENITIV(1, "Genitiv", "bez", "koho? čeho?"),
    DATIV(2, "Dativ", "ke", "komu? čemu?"),
    AKUZATIV(3, "Akuzativ", "vidim", "koho? co?"),
    VOKATIV(4, "Vokativ", "", "-"),
    LOKAL(5, "Lokál", "o", "(o) kom? (o) čem?"),
    INSTRUMENTAL(6, "Instrumentál", "s", "kým? čím?");

    companion object {
        @JvmStatic fun fromIndex(index: Int): CzechCase = entries.firstOrNull { it.index == index }
            ?: throw IllegalArgumentException("Invalid case index: $index. Valid range is 0-6.")
    }
}
